R/stability_selection.R

```diff
@@ -218,36 +226,48 @@
 #'
 #' @importFrom stabs stabsel
 #' @importFrom SummarizedExperiment SummarizedExperiment
+#' @importFrom cli cli_abort
 #'
 #'@export
 randLassoStabSel <- function(x, y, weakness=0.8, cutoff=0.8, PFER=2,
-                             mc.cores=1L, ...) {
+                             mc.cores=1L, glmnet.args = list(), ...) {
 
     # checks
     if (!is(x, "matrix")) {
-        stop("'x' must be a matrix")
+        cli_abort("{.arg x} must be a {.cls matrix}")
     }
     .assertVector(y, type = "numeric")
     if (nrow(x) != length(y)) {
-        stop("nrow of 'x' and length of 'y' are not equal. The rows of
-             x must be the same length and order as the elements in 'y'.")
+        cli_abort(paste0(
+        "nrow of {.arg x} and length of {.arg y} are not equal. The rows of ",
+        "{.arg x} must be the same length and order as the elements in {.arg y}."))
     }
     if (!is.null(names(y)) && !is.null(rownames(x)) &&
         !all(names(y) == rownames(x))) {
-        stop("'x' and 'y' have different names. Make sure that the names are
-             identical and that the orders match.")
+        cli_abort(paste0(
+            "{.arg x} and {.arg y} have different names. Make sure that the names are ",
+            "identical and that the orders match."))
     }
     if (is.null(rownames(x))) {
         rownames(x) <- paste0("obs", seq_len(nrow(x)))
     }
     if (is.null(colnames(x))) {
         colnames(x) <- paste0("pred", seq_len(ncol(x)))
     }
-
+    .assertVector(x = glmnet.args, type = "list")
+    if (length(glmnet.args) > 0) {
+        .assertVector(x = names(glmnet.args), type = "character")
+        if (any(i <- names(glmnet.args) %in% c("x", "y", "q", "weakness"))) {
+            warning(paste0("Ignoring the following elements of 'glmnet.args' (as ", 
+                           "they are set automatically): ", 
+                           paste(names(glmnet.args)[i], collapse = ",")))    
```

Comment:
maybe here the check should be that they are valid arguments in `glmnet::glmnet`

Reply:
I added this check